[MA-38] Added filter/validation queries for wpcids of active keywords

jellyvora-reachlocal opened the pull request:
Build: https://jenkins.media.dev.usa.reachlocalservices.com/view/media-automation/job/media-automation-build/68/

balkrishnada commented on src/main/resources/sql/removekeywords/validKeywordsWebPublisherCampaignIds.properties:
```diff
@@ -0,0 +1,49 @@
+  ON WPCHK.WebPublisherCampaign_idWebPublisherCampaign_FK = WPC.idWebPublisherCampaign \
+JOIN Offer o \
+  ON o.idOffer = WPC.Offer_idOffer_FK \
+WHERE SC.Campaign_idCampaign_FK = :campaignId \
```
Doing more research with the tables, it looks we should include Match Type as well in the condition to be sure enough. Let me confirm with Jonathan what does a NULL match type indicate in the BI table. I think they should match to 'Broad`.

balkrishnada commented on src/main/java/com/reachlocal/media/automation/jobs/removekeywords/webpublishercampaignkeyword/WebPublisherCampaignKeywordRepository.java:
```diff
@@ -0,0 +1,88 @@
+        this.wpcIdsHavingGeoKeywordInUpdateQuery = wpcIdsHavingGeoKeywordInUpdateQuery;
+    }
+
+    public List<WebPublisherCampaignKeyword> getActiveKeywordWebPublisherCampaignIds(Long campaignId, Long keywordId) {
```
`getWebPublisherCampaignIdsForActiveKeywords`

balkrishnada commented on src/main/java/com/reachlocal/media/automation/jobs/removekeywords/webpublishercampaignkeyword/WebPublisherCampaignKeywordRepository.java:
```diff
@@ -0,0 +1,88 @@
+        return jdbcTemplate.query(wpcIdsHavingActiveKeywordQuery, params, this::webPublisherCampaignKeywordRowMapper);
+    }
+
+    public List<WebPublisherCampaignKeyword> getActiveGeoKeywordWebPublisherCampaignIds(Long campaignId,
```
`getWebPublisherCampaignIdsForActiveGeoKeywords`

jellyvora-reachlocal commented on src/main/java/com/reachlocal/media/automation/jobs/removekeywords/auditlogs/AuditLog.java:
```diff
@@ -1,4 +1,6 @@
+
+import com.reachlocal.media.automation.jobs.removekeywords.keywordupdate.KeywordMatchType;
 
 public class AuditLog {
```
we need to have a column in low_qs_keyword_deletion_audit_log for `AdGroupId`

jellyvora-reachlocal replied:
I will add the column in V1__init sql file

balkrishnada commented on src/main/java/com/reachlocal/media/automation/jobs/removekeywords/webpublishercampaignkeyword/WebPublisherCampaignKeywordRepository.java:
```diff
@@ -0,0 +1,88 @@
+                                  this::webPublisherCampaignKeywordRowMapper);
+    }
+
+    public List<Long> getWebPublisherCampaignIdsInKeywordUpdate(List<Long> wpcIds, Long keywordId) {
```
`getDeactivationKeywordsForWebPublisherCampaignIds`

jellyvora-reachlocal replied:
That is not what the method does, it gets wpc ids for which keyword deletion entry is made in update table.

balkrishnada replied:
That's exactly the name `getDeactivationKeywordsForWebPublisherCampaignIds` suggests? There is a hardcoded activation type = 2 being used. What other name you have in mind?

balkrishnada commented on src/main/java/com/reachlocal/media/automation/jobs/removekeywords/webpublishercampaignkeyword/WebPublisherCampaignKeywordRepository.java:
```diff
@@ -0,0 +1,88 @@
+        return jdbcTemplate.queryForList(wpcIdsHavingKeywordInUpdateQuery, params, Long.class);
+    }
+
+    public List<Long> getWebPublisherCampaignIdsInGeoKeywordUpdate(List<Long> wpcIds,
```
`getDeactivationGeoKeywordsForWebPublisherCampaignIds`

jellyvora-reachlocal replied:
same as above

jellyvora-reachlocal commented on src/main/resources/sql/removekeywords/validKeywordsWebPublisherCampaignIds.properties:
```diff
@@ -0,0 +1,49 @@
+  AND WPC.wpc_end_date IS NULL
+
+
+GET_WEB_PUBLISHER_CAMPAIGN_IDS_IN_KEYWORDS_UPDATE = \
```
> WPC_HAS_KEYWORD_UPDATE_VALIDATION

Do we need `VALIDATION`,  also it is a check against all wpcIds of a keyword

balkrishnada replied:
Good point, please feel free to tweak in line with the suggestion.

jellyvora-reachlocal commented:
> good so far. some minor update to query will be needed based on feedback I will share later.
> Apart from that the queries also look good.

Addressed all the comments so far and updated build

balkrishnada commented on src/main/java/com/reachlocal/media/automation/jobs/removekeywords/webpublishercampaignkeyword/WebPublisherCampaignKeywordRepository.java:
```diff
@@ -0,0 +1,90 @@
+        return jdbcTemplate.queryForList(wpcIdsForDeletedKeywordQuery, params, Long.class);
+    }
+
+    public List<Long> getWebPublisherCampaignIdsInForDeletedGeoKeyword(List<Long> wpcIds,
```
please remove `In`
